ga4gh/datamodel/reads.py

```diff
@@ -325,8 +343,17 @@ def getReadAlignments(self, referenceId=None, start=None, end=None):
         referenceId, start, end = self.sanitizeAlignmentFileFetch(
             referenceId, start, end)
         readAlignments = samFile.fetch(referenceId, start, end)
-        for readAlignment in readAlignments:
-            yield self.convertReadAlignment(readAlignment)
+        if self._filterReads:
+            for readAlignment in readAlignments:
+                for tag in readAlignment.tags:
```

Comment:
What is ``readAlignment.tags``? If it's a ``dict`` we should just be able to say 
```
if 'RG' in readAlignment.tags:
    if readAlignment.tags['RG'] == self.getLocalId():
          yield self.convertReadAlignment(readAlignment)
```